### PR TITLE
[akka-scala] template upgrade dependency version and refactor #7171

### PR DESCRIPTION
**Pull request by shijinkui**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by wing328** [comment]
@shijinkui thanks for the PR.

cc @ramzimaalej @clasnake @jimschubert

File: modules/swagger-codegen/src/main/resources/akka-scala/pom.mustache [hunk 1]
```diff
@@ -1,246 +1,238 @@
...
-        <scala-version>2.10.4</scala-version>
-        <json4s-jackson-version>3.2.11</json4s-jackson-version>
-        <json4s-ext-version>3.2.11</json4s-ext-version>
-        <spray-version>1.3.1</spray-version>
-        <akka-version>2.3.9</akka-version>
-        <joda-version>1.2</joda-version>
-        <joda-time-version>2.2</joda-time-version>
-        <swagger-core-version>1.5.15</swagger-core-version>
-        <maven-plugin.version>1.0.0</maven-plugin.version>
+  <properties>
```
[the remark this answers is not in the document]
Comment (shijinkui) [reply modules/swagger-codegen/src/main/resources/akka-scala/pom.mustache]: > Please update the dependencies in build.gradle and build.sbt as well

Done. But there is no build.gradle.mustache in akka-scala

[the remark this answers is not in the document]
Comment (ramzimaalej) [reply modules/swagger-codegen/src/main/resources/akka-scala/pom.mustache]: thanks!

**Comment by ramzimaalej** [comment]
LGTM

**Comment by wing328** [comment]
@ramzimaalej thanks for reviewing.

@shijinkui thanks for the enhancement.

**Pull request merged**